repos/fdbt-site/src/pages/api/selectCaps.ts

```diff
@@ -47,19 +46,42 @@ export default async (req: NextApiRequestWithSession, res: NextApiResponse): Pro
             return;
         }
 
-        if (!Number.isInteger(Number(cap))) {
+        if (capChoice === 'yes' && !Number.isInteger(Number(cap))) {
             throw Error(`Received invalid cap id: ${cap}`);
         }
 
-        const selectedCap = await getCapContent(Number(cap), noc);
+        const selectedCap = capChoice === 'yes' ? await getCapContent(Number(cap), noc) : undefined;
+        const ticket = getSessionAttribute(req, MATCHING_JSON_ATTRIBUTE);
+        const matchingJsonMetaData = getSessionAttribute(req, MATCHING_JSON_META_DATA_ATTRIBUTE);
+
+        // if in edit mode
+        if (ticket && matchingJsonMetaData) {
+            const updatedTicket = {
+                ...ticket,
+                cap: capChoice === 'yes' && !!selectedCap ? { id: selectedCap.id } : undefined,
+            };
+
+            await putUserDataInProductsBucketWithFilePath(updatedTicket, matchingJsonMetaData.matchingJsonLink);
+            updateSessionAttribute(req, CAPS_DEFINITION_ATTRIBUTE, undefined);
 
-        if (capChoice === 'yes' && !!selectedCap) {
-            updateSessionAttribute(req, CAPS_DEFINITION_ATTRIBUTE, {
-                id: selectedCap.id,
```

Comment:
Can we put the file back to how it was structured before, and just move the edit mode check higher up - this means we dont need to edit the flow so much.